pyproject.toml

```diff
@@ -1,8 +1,88 @@
+[project]
+name = "scikit-learn"
+version = "1.5.dev0"
+description = "A set of python modules for machine learning and data mining"
+readme = "README.rst"
+dependencies = [
+  "numpy>=1.19.5",
+  "scipy>=1.6.0",
+  "joblib>=1.2.0",
+  "threadpoolctl>=2.0.0",
+]
+requires-python = ">=3.9"
+license = {text = "new BSD"}
+classifiers=[
+  "Intended Audience :: Science/Research",
+  "Intended Audience :: Developers",
+  "License :: OSI Approved :: BSD License",
+  "Programming Language :: C",
+  "Programming Language :: Python",
+  "Topic :: Software Development",
+  "Topic :: Scientific/Engineering",
+  "Development Status :: 5 - Production/Stable",
+  "Operating System :: Microsoft :: Windows",
+  "Operating System :: POSIX",
+  "Operating System :: Unix",
+  "Operating System :: MacOS",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+]
+
+[project.optional-dependencies]
+build = ["numpy>=1.19.5", "scipy>=1.6.0", "cython>=3.0.8", "meson-python>=0.15.0"]
+install = ["numpy>=1.19.5", "scipy>=1.6.0", "joblib>=1.2.0", "threadpoolctl>=2.0.0"]
+benchmark = ["matplotlib>=3.3.4", "pandas>=1.1.5", "memory_profiler>=0.57.0"]
+docs = [
+    "matplotlib>=3.3.4",
+    "scikit-image>=0.17.2",
+    "pandas>=1.1.5",
+    "seaborn>=0.9.0",
+    "memory_profiler>=0.57.0",
+    "sphinx>=6.0.0",
+    "sphinx-copybutton>=0.5.2",
+    "sphinx-gallery>=0.15.0",
+    "numpydoc>=1.2.0",
+    "Pillow>=7.1.2",
+    "pooch>=1.6.0",
+    "sphinx-prompt>=1.3.0",
+    "sphinxext-opengraph>=0.4.2",
+    "plotly>=5.14.0",
+]
+examples = [
+    "matplotlib>=3.3.4",
+    "scikit-image>=0.17.2",
+    "pandas>=1.1.5",
+    "seaborn>=0.9.0",
+    "pooch>=1.6.0",
+    "plotly>=5.14.0",
+]
+tests = [
+    "matplotlib>=3.3.4",
+    "scikit-image>=0.17.2",
+    "pandas>=1.1.5",
+    "pytest>=7.1.2",
+    "pytest-cov>=2.9.0",
+    "ruff>=0.2.1",
+    "black>=23.3.0",
+    "mypy>=1.3",
+    "pyamg>=4.0.0",
+    "polars>=0.19.12",
+    "pyarrow>=12.0.0",
+    "numpydoc>=1.2.0",
+    "pooch>=1.6.0",
+]
+maintenance = ["conda-lock>=2.4.2"]
```

Comment:
Could you expand `test_min_dependencies_pyproject_toml` to also check minimum version numbers for all of the optional dependency groups?

Reply:
I propose to add a test for optional dependencies and think about how to automatically the `pyproject.toml` in a more convenient way in a separate PR.

Medium-term, if I manage to take some time to look at `spin` (and convince people it is worth using it), this kind of maintenance tasks could be implemented as spin custom command.

Full disclosure: I did generate the optional dependencies from `_min_dependencies.py` `tag_to_packages`

Reply:
My last commit d2402e27b6 added this additional test (and found a few issues in the process e.g. cython 3.0.8 instead of 3.0.9)